src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs

```diff
@@ -64,6 +64,12 @@ internal class ConnectionManager : IConnectionManager {
             _userConnections = new ConcurrentDictionary<Uri, IConnection>(userConnections);
 
             UpdateRecentConnections();
+
+            _shell.Idle += OnIdle;
+        }
+
+        private void OnIdle(object sender, EventArgs e) {
```

Comment:
No, it just public. The other option is to create a lightweight package that will always load and use it to verify if VS startup is completed or not. I've sent a PR

---
In reply to: [78469414](https://github.com/Microsoft/RTVS/pull/2296#discussion_r78469414) [](ancestors = 78469414)